ComputerVision/Rest/ExtractText/Program.cs

```diff
@@ -0,0 +1,217 @@
+                Console.WriteLine("\nWait a moment for the results to appear.\n");
+                ExtractTextFromStreamAsync(imageFilePath).Wait();
+            }
+            else
+            {
+                Console.WriteLine("\nInvalid file path");
+            }
+
+            if (Uri.IsWellFormedUriString(remoteImageUrl, UriKind.Absolute))
+            {
+                ExtractTextFromUrlAsync(remoteImageUrl).Wait();
+            }
+            else
+            {
+                Console.WriteLine("\nInvalid remote image url:\n{0} \n", remoteImageUrl);
+            }
+
+            Console.WriteLine("\nPress Enter to exit...");
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// Gets the text from the specified image file by using
+        /// the Computer Vision REST API.
+        /// </summary>
+        /// <param name="imageFilePath">The image file with text.</param>
+        static async Task ExtractTextFromStreamAsync(string imageFilePath)
+        {
+            try
+            {
+                HttpClient client = new HttpClient();
+
+                // Request headers.
+                client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
+
+                // Request parameter.
+                string requestParameters = "mode=Handwritten";  // replace mode=Printed
+
+                // Assemble the URI for the REST API method.
+                string uri = uriBase + "?" + requestParameters;
+
+                HttpResponseMessage response;
+
+                // Two REST API methods are required to extract handwritten or printed text.
+                // One method to submit the image for processing, the other method
+                // to retrieve the text found in the image.
+
+                // operationLocation stores the URI of the second REST API method,
+                // returned by the first REST API method.
+                string operationLocation;
+
+                // Reads the contents of the specified local image
+                // into a byte array.
+                byte[] byteData = GetImageAsByteArray(imageFilePath);
+
+                // Adds the byte array as an octet stream to the request body.
+                using (ByteArrayContent content = new ByteArrayContent(byteData))
+                {
+                    // This example uses the "application/octet-stream" content type.
+                    // The other content types you can use are "application/json"
+                    // and "multipart/form-data".
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+
+                    // The first REST API method, Batch Read, starts
+                    // the async process to analyze the written text in the image.
+                    response = await client.PostAsync(uri, content);
+                }
+
+                // The response header for the Batch Read method contains the URI
```

Comment:
We should remove duplicate code between this case and the remote case. Let's move all the code between here and the rest of this function to another method that gets called from both cases. Something like WaitForExtractTextOperationResultAsync(HttpResponseMessage response).